### PR TITLE
Websocket: Update To Unsafe Signal for Close Rather than Pure Signal

### PR DESCRIPTION
**Pull request by ChristopherDavenport**

### CODE REVIEW

File: blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala [hunk 1]
```diff
@@ -80,18 +74,20 @@ class Http4sWSStage[F[_]](ws: ws4s.Websocket[F])(implicit F: Effect[F], val ec:
     val onStreamFinalize: F[Unit] =
       for {
         dec <- F.delay(count.decrementAndGet())
-        _ <- deadSignal.map(signal => if (dec == 0) signal.set(true))
+        _ <- if (dec == 0) deadSignal.set(true) else ().pure[F]
```
Comment (SystemFw) [inline blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala]: `().pure[F]` `-->` `F.unit`?

nicer, but utterly unimportant ofc

File: blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala [hunk 2]
```diff
@@ -80,18 +74,20 @@ class Http4sWSStage[F[_]](ws: ws4s.Websocket[F])(implicit F: Effect[F], val ec:
...
-    val wsStream = for {
-      dead <- deadSignal
-      in = inputstream.to(ws.receive).onFinalize(onStreamFinalize)
-      out = ws.send.onFinalize(onStreamFinalize).to(snk).drain
-      merged <- in.mergeHaltR(out).interruptWhen(dead).onFinalize(sendClose).compile.drain
-    } yield merged
+    val wsStream = inputstream
+      .to(ws.receive)
+      .onFinalize(onStreamFinalize)
+      .mergeHaltR(ws.send.onFinalize(onStreamFinalize).to(snk).drain)
```
Comment (SystemFw) [inline blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala]: `mergeHaltR(foo.drain)` can probably be `concurrently(foo)` now

**Comment by SystemFw** [comment]
Left a few style comments, but nothing too important. I'm also assuming this has been tested

File: blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala (hunk 2 above)
Comment (SystemFw) [reply blaze-core/src/main/scala/org/http4s/blazecore/websocket/Http4sWSStage.scala]: Actually scratch that sorry, you nee the output of `foo`, but want to complete when `bar` is done

**Comment by SystemFw** [comment]
@ChristopherDavenport you were too fast!

**Comment by SystemFw** [comment]
so I was imprecise, `concurrently` replaces `mergeHaltL` not `mergeHaltR`, so if you indeed want to finish the right branch finishes (but still return the output of the right branch), you indeed need what you used to have (sorry :grin:)

**Comment by ChristopherDavenport** [comment]
:(

**Pull request merged**